x/evm/keeper/integration_test.go

```diff
@@ -1,246 +1,252 @@
-		simutils.NewAppOptionsWithFlagHome(app.DefaultNodeHome),
-		baseapp.SetChainID(chainID),
-		baseapp.SetMinGasPrices(localMinGasPricesStr),
-	)
-
-	genesisState := app.NewTestGenesisState(newapp.AppCodec())
-	genesisState[types.ModuleName] = newapp.AppCodec().MustMarshalJSON(types.DefaultGenesisState())
-
-	stateBytes, err := json.MarshalIndent(genesisState, "", "  ")
-	s.Require().NoError(err)
-
-	// Initialize the chain
-	newapp.InitChain(
-		abci.RequestInitChain{
-			ChainId:         chainID,
-			Validators:      []abci.ValidatorUpdate{},
-			AppStateBytes:   stateBytes,
-			ConsensusParams: app.DefaultConsensusParams,
-		},
-	)
-
-	s.app = newapp
-	s.SetupApp(false, chainID)
-}
+		It("performs a transfer transaction", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			receiver := s.keyring.GetKey(1)
+			txArgs := evmtypes.EvmTxArgs{
+				To:     &receiver.Addr,
+				Amount: big.NewInt(1000),
+				// Hard coded gas limit to avoid failure on gas estimation because
+				// of the param
+				GasLimit: 100000,
+			}
+
+			res, err := s.factory.ExecuteEthTx(senderPriv, txArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
+		})
 
-func getNonce(addressBytes []byte) uint64 {
-	return s.app.EvmKeeper.GetNonce(
-		s.ctx,
-		common.BytesToAddress(addressBytes),
-	)
-}
+		It("fails when trying to perform contract deployment", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			constructorArgs := []interface{}{"coin", "token", uint8(18)}
+			compiledContract := contracts.ERC20MinterBurnerDecimalsContract
+			contractAddr, err := s.factory.DeployContract(
+				senderPriv,
+				compiledContract,
+				constructorArgs...,
+			)
+			Expect(err).NotTo(BeNil())
+			Expect(contractAddr).To(Equal(common.Address{}))
+		})
+	})
 
-func buildEthTx(
-	priv *ethsecp256k1.PrivKey,
-	to *common.Address,
-	gasLimit uint64,
-	gasPrice *big.Int,
-	gasFeeCap *big.Int,
-	gasTipCap *big.Int,
-	accesses *ethtypes.AccessList,
-) *evmtypes.MsgEthereumTx {
-	chainID := s.app.EvmKeeper.ChainID()
-	from := common.BytesToAddress(priv.PubKey().Address().Bytes())
-	nonce := getNonce(from.Bytes())
-	data := make([]byte, 0)
-	ethTxParams := &evmtypes.EvmTxArgs{
-		ChainID:   chainID,
-		To:        to,
-		Nonce:     nonce,
-		GasLimit:  gasLimit,
-		GasPrice:  gasPrice,
-		GasTipCap: gasTipCap,
-		GasFeeCap: gasFeeCap,
-		Accesses:  accesses,
-		Input:     data,
-	}
-	msgEthereumTx := evmtypes.NewTx(ethTxParams)
-	msgEthereumTx.From = from.String()
-	return msgEthereumTx
-}
+	When("EnableCall param is set to false", Ordered, func() {
+		BeforeAll(func() {
+			// Set params to default values
+			defaultParams := evmtypes.DefaultParams()
+			defaultParams.EnableCall = false
+			err := s.network.UpdateEvmParams(defaultParams)
+			Expect(err).To(BeNil())
+
+			err = s.network.NextBlock()
+			Expect(err).To(BeNil())
+		})
+
+		It("fails when performing a transfer transaction", func() {
```

Comment:
not sure why do we need to test a regular transfer here

Reply:
Because thats what we are actually blocking, anything besides contract deployment. So I was testing that both those types of transactions fail. Basically if that param is enabled and the `tx.to != nil` the transaction should fail.